public/app/features/alerting/NewAlertNotificationPage.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { PureComponent } from 'react';
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { NavModel, SelectableValue } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Form } from '@grafana/ui';
+import Page from 'app/core/components/Page/Page';
+import { NewNotificationChannelForm } from './components/NewNotificationChannelForm';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { createNotificationChannel, loadNotificationTypes, testNotificationChannel } from './state/actions';
+import { NotificationChannel, NotificationChannelDTO, StoreState } from '../../types';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  navModel: NavModel;
+  notificationChannels: NotificationChannel[];
+}
+
+interface DispatchProps {
+  createNotificationChannel: typeof createNotificationChannel;
+  loadNotificationTypes: typeof loadNotificationTypes;
+  testNotificationChannel: typeof testNotificationChannel;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+const defaultValues: NotificationChannelDTO = {
+  name: '',
+  type: { value: 'email', label: 'Email' },
+  sendReminder: false,
+  disableResolveMessage: false,
+  frequency: '15m',
+  settings: {
+    uploadImage: config.rendererAvailable,
+    autoResolve: true,
+    httpMethod: 'POST',
+    severity: 'critical',
+  },
+  isDefault: false,
+};
+
+class NewAlertNotificationPage extends PureComponent<Props> {
+  componentDidMount() {
+    this.props.loadNotificationTypes();
+  }
+
+  onSubmit = (data: NotificationChannelDTO) => {
+    /*
+      Some settings can be options in a select, in order to not save a SelectableValue<T>
+      we need to use check if it is a SelectableValue and use its value.
+    */
+    const settings = Object.fromEntries(
+      Object.entries(data.settings).map(([key, value]) => {
+        return [key, value.hasOwnProperty('value') ? value.value : value];
+      })
+    );
+
+    this.props.createNotificationChannel({
+      ...defaultValues,
+      ...data,
+      type: data.type.value,
+      settings: { ...Object.assign(defaultValues.settings, settings) },
```

Comment:
`Object.assign` will **[copy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)** own enumerable props from `settings`, to `defaultValues.settings`. This could lead to subtle bugs, since you are reusing `defaultValues`. 

You'd probably want to leave `defaultValues` unchanged during the component lifetime. If that is true, than a better approach would be to copy both to an empty object, like in:
```suggestion
      settings: Object.assign({}, defaultValues.settings, settings) },
```

Reply:
Agree with @atymchuk ☝️